pageserver/src/pgdatadir_mapping.rs

```diff
@@ -303,7 +304,11 @@ impl Timeline {
         ctx: &RequestContext,
     ) -> Result<LsnForTimestamp, PageReconstructError> {
         let gc_cutoff_lsn_guard = self.get_latest_gc_cutoff_lsn();
-        let min_lsn = *gc_cutoff_lsn_guard;
+        // We use this method to figure out the branching LSN for new branch, but
+        // GC cutoff could be before branching point and we cannot create new branch
+        // with LSN < `ancestor_lsn`. Thus, pick the maximum of these two just to be
+        // on the safe side.
+        let min_lsn = std::cmp::max(*gc_cutoff_lsn_guard, self.get_ancestor_lsn());
```

Comment:
> Makes sense. Do I understand correctly, you would currently get an error in that case, and now you get 'past'?

As we previously noticed `gc_cutoff_lsn` could be before `ancestor_lsn`, so theoretically we can pass such timestamp, so that we will find commit with LSN < `ancestor_lsn` and branch creation will fail

So I actually thought that we could potentially get `present` and now we will get `past`

> Please add a test case for that, if we didn't have one already.

Not sure how to properly test it, as it depends on `gc_cutoff_lsn`